### ISSUE TITLE
FroSch / boost-related errors in Albany nightly tests

### ISSUE THREAD

**Issue by ikalash** [sandialabs/Albany#758]
There are FroSch / boost-related errors in the Albany nightly tests:

https://sems-cdash-son.sandia.gov/cdash/viewBuildError.php?buildid=24752

```
/.../repos/Albany/src/disc/stk/Albany_IossSTKMeshStruct.cpp(484): error: namespace "boost" has no member "any"
```
```
/home/projects/albany/nightlyCDashTrilinosBlake/build/TrilinosSerialInstall/include/FROSch_InterfaceEntity_decl.hpp(95): error: "Vector" is ambiguous
```
The CDash site was down for maintenance starting last Thursday or so, and I was on vacation earlier this week, so I'm not 100% sure when it started.  Anyone know what happened?

**Comment by mperego** [comment]
Not a nice welcome back from vacation!
@jewatkins and I were looking into that yesterday. 
The boost issue is due to a snapshot of STK into Trilinos. They removed some deprecated functions that used boost and now we need to modify Albany code to comply with it. I spent an hour yesterday trying to fix this but it turned out to be a bit more involved than I expected.

The second issue is likely related to a snapshot of Kokkos that went in on Friday and probably it need to be fixed in Trilinos or FROSch, however I haven't had time to look into that carefully. I sent an email to Alexander, I'll let you know what he thinks.

**Comment by mperego** [comment]
update: FROSch team is working on it. We might have a fix soon.

**Comment by jfike** [comment]
We are seeing the Vector is ambiguous error in another application code.  I was able to resolve this by making a few minor changes in kokkos-kernels.  I have a ticket in to kokkos https://github.com/kokkos/kokkos/issues/4497

**Comment by mperego** [comment]
Thanks @jfike. My understanding is that @iyamazaki  was able to fix the problem in FROSch specifying the namespace Xpetra, but what you propose might fix FROSch as well.

**Comment by ikalash** [comment]
> Not a nice welcome back from vacation! 

@mperego : happens every time.  Everything broke in the 2 days I was out, and I mean everything.... :(

**Comment by mperego** [comment]
The FROSch error should be fixed by PR  trilinos/Trilinos#9903 and the STK error by PR #760.
